src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/MessagePageRenderer.cs

```diff
@@ -0,0 +1,145 @@
+﻿/*
+ Copyright Airship and Contributors
+*/
+
+using System;
+using Foundation;
+using UIKit;
+using UrbanAirship.NETStandard.MessageCenter;
+using WebKit;
+using Xamarin.Forms.Platform.iOS;
+
+namespace UrbanAirship.NETStandard.iOS
+{
+    public class MessagePageRenderer : PageRenderer, IWKNavigationDelegate, IUANativeBridgeDelegate
+    {
+        private UAWebView webView;
+        private UANativeBridge nativeBridge;
+        private MessagePage messagePage;
+        private string messageId;
+
+        public MessagePageRenderer()
+        {
+            webView = new UAWebView();
+            nativeBridge = UANativeBridge.NativeBridge();
+            webView.NavigationDelegate = nativeBridge;
+            nativeBridge.ForwardNavigationDelegate = this;
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+
+            webView.Frame = UIScreen.MainScreen.Bounds;
+            webView.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
+            webView.TranslatesAutoresizingMaskIntoConstraints = true;
+
+            View.AddSubview(webView);
+        }
+
+        public void LoadMessage(string messageId)
+        {
+            var message = UAMessageCenter.Shared().MessageList.Message(messageId);
+            if (message != null)
+            {
+                LoadMessageBody(message);
+            }
+            else
+            {
+                UAMessageCenter.Shared().MessageList.RetrieveMessageList(() =>
+                {
+                    message = UAMessageCenter.Shared().MessageList.Message(messageId);
+                    if (message != null && !message.IsExpired())
+                    {
+                        LoadMessageBody(message);
+                    }
+                    else
+                    {
+                        messagePage.OnRendererLoadFailed(messageId, false, MessageFailureStatus.Unavailable);
+                    }
+                }, () =>
+                {
+                    messagePage.OnRendererLoadFailed(messageId, false, MessageFailureStatus.FetchFailed);
+                });
+            }
+        }
+
+        public void Close()
+        {
+            messagePage.OnRendererClosed(messageId);
+        }
+
+        [Export("webView:decidePolicyForNavigationResponse:decisionHandler:")]
```

Comment:
These export annotations were added by VS automatically at its own suggestion, and my assumption is that they're needed because the bound delegate protocol diverges significantly from the original selector and messes up introspection on the Obj-c side. But that's just a guess.